bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java

```diff
@@ -0,0 +1,384 @@
+                    "Globe must have a valid bridge selected before it can come online.");
+            return;
+        } else {
+            globeType = thing.getThingTypeUID().getId();// eg rgb_cct
+            String globeLocation = this.getThing().getUID().getId();// eg 0x014
+            remotesGroupID = globeLocation.substring(globeLocation.length() - 1, globeLocation.length());// eg 4
+            String remotesIDCode = globeLocation.substring(0, globeLocation.length() - 1);// eg 0x01
+            fullCommandTopic = COMMANDS_BASE_TOPIC + remotesIDCode + "/" + globeType + "/" + remotesGroupID;
+            fullStatesTopic = STATES_BASE_TOPIC + remotesIDCode + "/" + globeType + "/" + remotesGroupID;
+            // Need to remove the lowercase x from 0x12AB in case it contains all numbers
+            String caseCheck = globeLocation.substring(2, globeLocation.length() - 1);
+            if (!caseCheck.equals(caseCheck.toUpperCase())) {
+                logger.warn(
+                        "The milight globe {}{} is using lowercase for the remote code when the hub needs UPPERCASE",
+                        remotesIDCode, remotesGroupID);
+            }
+            channelPrefix = BINDING_ID + ":" + globeType + ":" + localBridge.getUID().getId() + ":" + remotesIDCode
+                    + remotesGroupID + ":";
+            connectMQTT();
+        }
+    }
+
+    private void sendMQTT(String payload) {
+        MqttBrokerConnection localConnection = connection;
+        if (localConnection != null) {
+            localConnection.publish(fullCommandTopic, payload.getBytes(), 1, false);
+        }
+    }
+
+    @Override
+    public void processMessage(String topic, byte[] payload) {
+        String state = new String(payload, StandardCharsets.UTF_8);
+        logger.trace("Recieved the following new Milight state:{}:{}", topic, state);
+        processIncomingState(state);
+    }
+
+    @Override
+    public void connectionStateChanged(MqttConnectionState state, @Nullable Throwable error) {
+        logger.debug("MQTT brokers state changed to:{}", state);
+        switch (state) {
+            case CONNECTED:
+                updateStatus(ThingStatus.ONLINE);
+                break;
+            case CONNECTING:
+            case DISCONNECTED:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Bridge (broker) is not connected to your MQTT broker.");
+        }
+    }
+
+    public void connectMQTT() {
+        Bridge localBridge = this.getBridge();
+        if (localBridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED,
+                    "Bridge is missing or offline, you need to setup a working MQTT broker first.");
+            return;
+        }
+        ThingUID thingUID = new ThingUID(new ThingTypeUID("mqtt", "broker"), localBridge.getUID().getId());
```

Comment:
Why are you hardcoding the broker thing name here? Isn't the broker accessible through the bridge thing?

Reply:
FIXED.